webhook_server/utils/helpers.py

```
@@ -335,12 +341,54 @@ async def run_command(
+                            try:
+                                sub_process.kill()
+                                await sub_process.wait()
+                            except Exception:
+                                pass
+                except Exception:
+                    pass  # Process may already be dead
             return False, "", f"Command timed out after {timeout}s"
+        except Exception as comm_exc:
+            # If communicate() raises any exception (not just TimeoutError),
+            # we need to ensure the subprocess is cleaned up
+            logger.error(f"{log_prefix} Command '{logged_command}' communicate() raised exception: {comm_exc}")
```

Ruff (0.14.5)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)